### PR TITLE
feat(command-bar-reflow): create CommandBarButtonsMenu component

### PR DESCRIPTION
**Pull request by pownkel**
#### Description of changes

Create component for ... button (not currently used) with menu that currently renders the command bar buttons as-is:
elipses menu

#### Pull request checklist

- [x] Addresses an existing issue: #1739635
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/DetailsView/components/command-bar-buttons-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// Licensed under the MIT License.
+import { NamedFC } from 'common/react/named-fc';
+import { CommandBarProps } from 'DetailsView/components/details-view-command-bar';
+import { CommandBarButton, IOverflowSetItemProps, OverflowSet } from 'office-ui-fabric-react';
+import * as React from 'react';
+
+export type CommandBarButtonsMenuProps = CommandBarProps;
+
+export const CommandBarButtonsMenu = NamedFC<CommandBarButtonsMenuProps>(
+    'DetailsListIssuesView',
```
Comment (waabid) [inline src/DetailsView/components/command-bar-buttons-menu.tsx:11]: Should probably be `CommandBarButtonsMenu`

File: src/DetailsView/components/command-bar-buttons-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            item.onRender(props);
+        };
+
+        const onRenderOverflowButton = (overflow: IOverflowSetItemProps[]): JSX.Element => {
+            return (
+                <CommandBarButton
+                    ariaLabel="More items"
+                    role="menuitem"
+                    menuIconProps={{ iconName: 'More' }}
+                    menuProps={{ items: overflow! }}
```
Comment (waabid) [inline src/DetailsView/components/command-bar-buttons-menu.tsx:23]: What's the `!` for?

File: src/DetailsView/components/command-bar-buttons-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import { CommandBarProps } from 'DetailsView/components/details-view-command-bar';
+import { CommandBarButton, IOverflowSetItemProps, OverflowSet } from 'office-ui-fabric-react';
+import * as React from 'react';
+
+export type CommandBarButtonsMenuProps = CommandBarProps;
+
+export const CommandBarButtonsMenu = NamedFC<CommandBarButtonsMenuProps>(
+    'DetailsListIssuesView',
+    props => {
+        const onRenderItem = (item: IOverflowSetItemProps) => {
```
Comment (waabid) [inline src/DetailsView/components/command-bar-buttons-menu.tsx:13]: So ... onRenderItem doesn't return anything? A bit confused about how this works ...

File: src/DetailsView/components/command-bar-buttons-menu.tsx (hunk 2 above)
Comment (pownkel) [reply src/DetailsView/components/command-bar-buttons-menu.tsx:23]: It's a non-null assertion. It came from some code I copied from the office fabric docs, but I can take it out.

File: src/DetailsView/components/command-bar-buttons-menu.tsx (hunk 3 above)
Comment (pownkel) [reply src/DetailsView/components/command-bar-buttons-menu.tsx:13]: Not sure how it was working before, but I added the return statement

Comment (lisli1) [reply src/DetailsView/components/command-bar-buttons-menu.tsx:13]: Should we also include the function return type here, as in onRenderOverflowButton?

File: src/tests/unit/tests/DetailsView/components/command-bar-buttons-menu.test.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            .setup(s => s(commandBarButtonsMenuProps))
+            .returns(() => <></>)
+            .verifiable(Times.once());
+
+        const wrapper = shallow(<CommandBarButtonsMenu {...commandBarButtonsMenuProps} />);
+        const renderedProps = wrapper.getElement().props;
+        const overflowItems: IOverflowSetItemProps[] = renderedProps.overflowItems;
+        const onRenderItem: (props: IOverflowSetItemProps) => JSX.Element =
+            renderedProps.onRenderItem;
+        expect(overflowItems).not.toBeUndefined();
```
Comment (lisli1) [inline src/tests/unit/tests/DetailsView/components/command-bar-buttons-menu.test.tsx:55]: Can we use .toBeDefined() here?

File: src/tests/unit/tests/DetailsView/components/command-bar-buttons-menu.test.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            {
+                key: 'item 1',
+            },
+            {
+                key: 'item 2',
+            },
+        ];
+        const wrapper = shallow(<CommandBarButtonsMenu {...commandBarButtonsMenuProps} />);
+        const onRenderOverflowButton = wrapper.getElement().props.onRenderOverflowButton;
+        expect(onRenderOverflowButton).not.toBeUndefined();
```
Comment (lisli1) [inline src/tests/unit/tests/DetailsView/components/command-bar-buttons-menu.test.tsx:78]: Same here; could we use toBeDefined()?

**Pull request merged**